File: lib/models/sprint_state.rb
require 'sinatra/activerecord'

class SprintState < ActiveRecord::Base
    include Obfuscate

    def arbiter_id
        encrypt self[:arbiter_id]
    end

    belongs_to :sprint
    belongs_to :state
    has_many :states
    has_many :contributors
    belongs_to :contributor
end

File: integrations-client/app/models/sprint-state.js
import DS from 'ember-data';

const { attr, Model } = DS;

export default DS.Model.extend({
    active_contribution_id: attr('number'),
    contributor_id: attr('number'),
    arbiter_id: attr('number'),
    sprint_id: DS.belongsTo('sprint'),
    state_id: DS.belongsTo('state'),
    sprint: attr('number'),
    state: attr('number'),
    deadline: attr('date'),
    sha: attr('string'),
    contributors: DS.hasMany('contributor'),
    created_at: attr('date'),
    updated_at: attr('date'),
    pull_request: attr('number'),
    merged: attr('boolean')
});
